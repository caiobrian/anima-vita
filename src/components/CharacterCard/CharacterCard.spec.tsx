import { render, screen } from '@testing-library/react'

import { CharacterCard } from '.'

const characterCardData = {
  name: 'Test Name',
  description: 'Test Description',
  imageUrl: 'https://test.com/test.jpg',
  otherNames: ['Test Other Name 1', 'Test Other Name 2']
}

const NO_DESCRIPTION_FOUND = 'No description found'
const NO_IMAGE_FOUND = 'No image found'

describe('CharacterCard', () => {
  it('should render correctly', () => {
    const { container } = render(<CharacterCard {...characterCardData} />)

    expect(container).toMatchSnapshot()
  })

  it('should render character name in front card when has no imageUrl', () => {
    render(<CharacterCard {...characterCardData} imageUrl="" />)

    const name = screen.getByTestId('character-card-name')
    expect(name).toHaveTextContent(characterCardData.name)
  })

  it('should render No Image Found when has no imageUrl', () => {
    render(<CharacterCard {...characterCardData} imageUrl="" />)

    const cardFront = screen.getByTestId('character-card-front')
    expect(cardFront).toHaveTextContent(NO_IMAGE_FOUND)
  })

  it('should not render character name or No Image Found in front card when has imageUrl', () => {
    render(<CharacterCard {...characterCardData} />)

    const name = screen.getByTestId('character-card-name')
    expect(name).toHaveTextContent('')

    const cardFront = screen.getByTestId('character-card-front')
    expect(cardFront).not.toHaveTextContent(NO_IMAGE_FOUND)
  })

  it('should not render the character description and name in front card', () => {
    render(<CharacterCard {...characterCardData} />)

    const frontCard = screen.getByTestId('character-card-front')
    expect(frontCard).not.toHaveTextContent(characterCardData.description)
    expect(frontCard).not.toHaveTextContent(characterCardData.name)
  })

  it('should render the character description and name in back card', () => {
    render(<CharacterCard {...characterCardData} />)

    const backCard = screen.getByTestId('character-card-back')
    expect(backCard).toHaveTextContent(characterCardData.description)
    expect(backCard).toHaveTextContent(characterCardData.name)
  })

  it('should render No Description Found when has no description', () => {
    render(<CharacterCard {...characterCardData} description="" />)

    const cardBack = screen.getByTestId('character-card-back')
    expect(cardBack).toHaveTextContent(NO_DESCRIPTION_FOUND)
  })

  it('should not render "Also know as:" when character has no other names', () => {
    render(<CharacterCard {...characterCardData} otherNames={[]} />)

    const otherNames = screen.queryByTestId('character-othernames')
    expect(otherNames).toBeEmptyDOMElement()
  })

  it('should format other character names with "Also know as:" when has other names', () => {
    render(<CharacterCard {...characterCardData} />)

    const othernames = characterCardData.otherNames.join(', ')
    const othernamesText = screen.getByText(`Also known as: ${othernames}`)

    expect(othernamesText).toBeInTheDocument()
  })
})
