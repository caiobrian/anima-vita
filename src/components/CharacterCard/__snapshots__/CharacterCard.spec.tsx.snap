// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard should render correctly 1`] = `
<div>
  <div
    class="sc-gsDKAQ jdiYNr"
  >
    <div
      class="sc-bdvvtL eizJli"
    >
      <div
        class="sc-dkPtRN sc-hKwDye bYWgbM kvxXbG"
        data-testid="character-card-front"
      >
        <h2
          data-testid="character-card-name"
        />
        <p />
      </div>
      <div
        class="sc-dkPtRN sc-eCImPb bYWgbM kYagot"
        data-testid="character-card-back"
      >
        <div
          class="sc-jRQBWg leOgDK"
        >
          <h2
            class="sc-gKclnd drgwOt"
          >
            Test Name
          </h2>
          <h3
            class="sc-iCfMLu hhyEnO"
            data-testid="character-othernames"
          >
            Also known as: Test Other Name 1, Test Other Name 2
          </h3>
          <p
            class="sc-furwcr cBpvSA"
          >
            Test Description
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
